Automattic/newspack-plugin
# feat(ia): Newspack > Settings > Social - part 2

### All Submissions:


### Changes proposed in this Pull Request:

Following from #3257, the PR introduces the X (Twitter) and Meta Pixels for Newspack > Settings > Social tab. This PR also introduces `WizardsToggleHeaderCard`:
* Easily create these specific action card layouts.
* Render props to allow consuming components to act on state values and use state callbacks.
* Abstract and unify common toggle on/off logic into reusable component.

**Additional:**
* Updated Twitter to X.
* Fixed an issue where Meta Pixel IDs would strip leading zeros during route validation callback.

### How to test the changes in this Pull Request:

1. Checkout this code and build assets `git checkout origin/feat/ia-setting-social__pixel && npm run build`
2. Navigate to _/wp-admin/admin.php?page=newspack-settings#/social_
3. Observe the below loads/exists:

![Screenshot 2024-07-25 at 09 07 11](https://github.com/user-attachments/assets/0ac10ba2-037f-4d4a-943b-4aaccffee970)

Once loaded - test the following scenarios:
* Ensure Toggling on/off works as expected, should show/hide form.
* Test Pixel ID inputs i.e. add a value and click "Save Settings":
   * Meta Pixel ID should only accept numeric values.
   * X Pixel ID should only accept alpha-numeric values.
   * Neither inputs should allow empty values.

### Other information:
